libcst/metadata/scope_provider.py

```diff
@@ -699,22 +700,66 @@ def visit_Attribute(self, node: cst.Attribute) -> Optional[bool]:
 
     def visit_Call(self, node: cst.Call) -> Optional[bool]:
         self.__top_level_attribute_stack.append(None)
+        if any(
+            qn.name == "typing.TypeVar"
+            for qn in self.scope.get_qualified_names_for(node)
+        ):
+            node.func.visit(self)
+            for arg in node.args[:1]:
+                arg.visit(self)
+            self.__in_annotation.add(node)
+            for arg in node.args[1:]:
+                arg.visit(self)
+            return False
+        return True
 
     def leave_Call(self, original_node: cst.Call) -> None:
         self.__top_level_attribute_stack.pop()
+        self.__in_annotation.discard(original_node)
 
     def visit_Annotation(self, node: cst.Annotation) -> Optional[bool]:
         self.__in_annotation.add(node)
 
     def leave_Annotation(self, original_node: cst.Annotation) -> None:
         self.__in_annotation.discard(original_node)
 
+    def visit_SimpleString(self, node: cst.SimpleString) -> Optional[bool]:
+        self._handle_string_annotation(node)
+        return False
+
+    def visit_ConcatenatedString(self, node: cst.ConcatenatedString) -> Optional[bool]:
+        self._handle_string_annotation(node)
+        return False
+
+    def _handle_string_annotation(
+        self, node: Union[cst.SimpleString, cst.ConcatenatedString]
+    ) -> None:
+        if self.__in_annotation and not self.__in_literal:
+            value = node.evaluated_value
+            if value:
+                mod = cst.parse_module(value)
+                mod.visit(self)
+
+    def visit_Subscript(self, node: cst.Subscript) -> Optional[bool]:
+        if any(
+            qn.name == "typing.Literal"
+            for qn in self.scope.get_qualified_names_for(node.value)
+        ):
+            node.value.visit(self)
+            self.__in_literal.add(node)
+            for s in node.slice:
+                s.visit(self)
+            return False
```

Comment:
Can `Literal` contains a name reference as string?
https://docs.python.org/3/library/typing.html#typing.Literal
It doesn't seem like it.

Reply:
Yes, we want to ignore the strings inside a `Literal`, so we ignore those by tracking if we're in a `Literal` using `__in_literal`.

Reply:
NVM, figured it out.

Reply:
```py
import foo
def function(param: Literal["foo", "bar"]) -> None:
    pass
```

If we don't ignore the strings inside `Literal`, in the above example the imported `foo` would have a wrong access attributed to it.